frame/pools/src/lib.rs

```diff
@@ -0,0 +1,1135 @@
+//! # Delegation pools for nominating
+//!
+//! This pallet allows delegators to delegate their stake to nominating pools, each of which acts as
+//! a nominator and nominates validators on their behalf.
+//!
+//! ## Design
+//!
+//! _Notes_: this section uses pseudo code to explain general design and does not necessarily
+//! reflect the exact implementation. Additionally, a strong knowledge of `pallet-staking`'s api is
+//! assumed.
+//!
+//! The delegation pool abstraction is composed of:
+//!
+//!  bonded pool: Tracks the distribution of actively staked funds. See [`BondedPool`] and
+//! [`BondedPools`]
+//! * reward pool: Tracks rewards earned by actively staked funds. See [`RewardPool`] and
+//!   [`RewardPooks`].
+//! * unbonding sub pools: This a collection of pools at different phases of the unbonding
+//!   lifecycle. See [`SubPools`] and [`SubPoolsStorage`].
+//! * delegators: Accounts that are members of pools. See [`Delegator`] and [`Delegators`].
+//! In order to maintain scalability, all operations are independent of the number of delegators. To
+//! do this, we store delegation specific information local to the delegator while the pool data
+//! structures have bounded datum .
+//!
+//! ### Design goals
+//!
+//! * Maintain integrity of slashing events, sufficiently penalizing delegators that where in the
+//!   pool while it was backing a validator that got slashed.
+//! * Maximize scalability in terms of delegator count.
+//!
+//! ### Bonded pool
+//!
+//! A bonded pool nominates with its total balance, excluding that which has been withdrawn for
+//! unbonding. The total points of a bonded pool are always equal to the sum of points of the
```

Comment:
what's a "point"?

Reply:
I see what it is in a few lines further, but this is again pointing me to ask for a terminology section where all the keywords are explained.